### PR TITLE
Allow requesting more than 65536 bytes in the browser

### LINKED ISSUE #14: requested too many random bytes
It looks like this library limits the number of random bytes you can request to [65536](https://github.com/crypto-browserify/randombytes/blob/master/browser.js#L18).  Node limits it to the [max value of an unsigned 32 bit integer](https://github.com/nodejs/node/blob/master/lib/internal/crypto/random.js#L52).

Is there any reason for the discrepancy?

### PR DESCRIPTION
**Pull request by achingbrain**
Calls `crypto.getRandomValues(buffer)` repeatedly until the request number of bytes have been generated.

Will throw an error if fewer than zero or more than the max value of a uint32 bytes are requested in the same way as node.js.

Fixes #14

### CODE REVIEW

File: browser.js [hunk 1]
```diff
@@ -26,13 +32,30 @@ function randomBytes (size, cb) {
...
-  var bytes = Buffer.from(rawBytes.buffer)
+  return Buffer.from(rawBytes.buffer)
+}
+
+function randomBytes (size, cb) {
+  if (size > MAX_UINT32) {
+    throw new Error('Requested too many random bytes - the maximum allowed are ' + MAX_UINT32)
+  }
+
+  var bytes = Buffer.alloc(0)
```
Comment (calvinmetcalf) [inline browser.js]: why are we making this empty buffer, can't we just make this undefined

File: browser.js [hunk 2]
```diff
@@ -26,13 +32,30 @@ function randomBytes (size, cb) {
...
+
+  if (size > MAX_BYTES) {
+    bytes = randomBytes(size - MAX_BYTES)
+
+    bytes = Buffer.concat([bytes, generateBytes(MAX_BYTES)])
+  } else if (size > 0) {
```
Comment (calvinmetcalf) [inline browser.js]: how about if size is less then or equal to zero then we make an empty buffer

File: browser.js [hunk 3]
```diff
@@ -26,13 +32,30 @@ function randomBytes (size, cb) {
...
+
+function randomBytes (size, cb) {
+  if (size > MAX_UINT32) {
+    throw new Error('Requested too many random bytes - the maximum allowed are ' + MAX_UINT32)
+  }
+
+  var bytes = Buffer.alloc(0)
+
+  if (size > MAX_BYTES) {
+    bytes = randomBytes(size - MAX_BYTES)
```
Comment (calvinmetcalf) [inline browser.js]: we should probably not do this recursively instead we should probably make all the sub buffers and concat them once, or create a single ArrayBuffer and pass views of MAX_BYTES or less so you don't need to even do a concat

File: browser.js (hunk 1 above)
Comment (achingbrain) [reply browser.js]: As you prefer.

File: browser.js (hunk 2 above)
Comment (achingbrain) [reply browser.js]: Node throws an error if `size` is less than zero.  I've added this check to this PR.

File: browser.js (hunk 3 above)
Comment (achingbrain) [reply browser.js]: As you prefer.

File: browser.js [hunk 4]
```diff
@@ -26,13 +32,47 @@ function randomBytes (size, cb) {
...
+    bytes = Buffer.alloc(size)
+
+    for (var read = 0; read < size;) {
+      var next = size - read
+
+      if (next > MAX_BYTES) {
+        next = MAX_BYTES
+      }
+
+      generateBytes(next).copy(bytes, read)
```
Comment (calvinmetcalf) [inline browser.js]: this is still creating multiple buffers and copying them around, if you create 1 buffer and then use the slice method it creates a view of the parent buffer without actually allocating a new one and changes to that view are reflected in the parent

**Comment by achingbrain** [comment]
@calvinmetcalf thanks for your input - were there any further comments or do you think this can be merged?

**Comment by calvinmetcalf** [comment]
I think this can be done with zero copies, let me do some tests

**Comment by calvinmetcalf** [comment]
hey take a look at #16, it turns out we can simplify a lot of stuff

**Comment by calvinmetcalf** [comment]
Yeah but we're not using the built in buffer, we're using safe buffer which
seems to work on phantom

**Comment by achingbrain** [comment]
Should probably remove that comment then?

**Comment by calvinmetcalf** [comment]
yeah that comment is gone in the other pull, which I'm going to close this in favor of

**Pull request closed without merging**